form/src/main/java/org/dhis2/form/data/DataEntryBaseRepository.kt

```diff
@@ -40,8 +41,20 @@ abstract class DataEntryBaseRepository(
         val item = when {
             fieldUiModel.optionSet != null -> {
                 fieldUiModel.apply {
-                    this.optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten()
-                    this.optionsToShow = optionsInGroupsToShow
+                    val optionsetConfig = optionSetConfiguration
+                    this.optionSetConfiguration = when (optionsetConfig) {
+                        is OptionSetConfiguration.BigOptionSet -> optionsetConfig.copy(
+                            optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten(),
+                            optionsToShow = optionsInGroupsToShow
+                        )
+                        is OptionSetConfiguration.DefaultOptionSet -> optionsetConfig.copy(
+                            optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten(),
+                            optionsToShow = optionsInGroupsToShow
+                        )
+                        null -> {
+                            optionSetConfiguration
+                        }
```

Comment:
If both optionSetConfiguration are doing the same, maybe a simple `if (optionsetConfig != null) { } else { }` would be enough to not duplicate code.

Reply:
Updated to a simpler methor inside the OptionSetConfigation sealed class.